### PR TITLE
feat: Add ruler and alertmanager

### LINKED ISSUE #534: enterprise-metrics: Tokengen should not be enabled by default.
It would be useful for tokengen to be split out for default manifests so that a user opts in to its roll out. Currently after initial roll out, a number of changes to the library configuration result in changes to immutable Job fields which can be confusing. Modelling around an explicit import better fits the pattern of a one-off job.

### PR DESCRIPTION
**Pull request by jdbaldry**
- resolves #541
- resolves #534

### CODE REVIEW

**Comment by jdbaldry** [comment]
@eamonryan I added you as a reviewer in case you wanted to test this out in your environment

**Review by jtlisi — APPROVED** [review]
LGTM

File: enterprise-metrics/docs/README.md [hunk 1]
```diff
@@ -297,6 +327,62 @@ $ kubectl create secret generic gem-license -from-file=license.jwt
 
 `service` is the Kubernetes Service for the admin-api.
 
+## obj alertmanager
+
+`alertmanager` has configuration for the alertmanager. To disable the alertmanager, ensure the ruler object field is hidden
```
Comment (MichelHollands) [inline enterprise-metrics/docs/README.md:332]: Should it say 'ruler object'?

File: enterprise-metrics/docs/README.md [hunk 2]
```diff
@@ -297,6 +327,62 @@ $ kubectl create secret generic gem-license -from-file=license.jwt
...
+
+## obj alertmanager.args
+
+`args` is a convenience field that can be used to modify the alertmanager container arguments as key-value pairs.
+
+
```
Comment (MichelHollands) [inline enterprise-metrics/docs/README.md:338]: Extra empty line here

File: enterprise-metrics/docs/README.md [hunk 3]
```diff
@@ -297,6 +327,62 @@ $ kubectl create secret generic gem-license -from-file=license.jwt
...
+### bool alertmanager.args.auth.type
+
+*Default value: * `enterprise`
+
+`auth.type` configures the type of authentication in use.
+`enterprise` uses Grafana Enterprise token authentication.
+`default` uses Cortex authentication.
+
+
```
Comment (MichelHollands) [inline enterprise-metrics/docs/README.md:359]: Extra empty line

File: enterprise-metrics/docs/README.md [hunk 4]
```diff
@@ -653,6 +751,61 @@ $ kubectl create secret generic gem-license -from-file=license.jwt
...
+### bool ruler.args.auth.type
+
+*Default value: * `enterprise`
+
+`auth.type` configures the type of authentication in use.
+`enterprise` uses Grafana Enterprise token authentication.
+`default` uses Cortex authentication.
+
+
```
Comment (MichelHollands) [inline enterprise-metrics/docs/README.md:776]: Extra empty line

File: enterprise-metrics/docs/README.md [hunk 5]
```diff
@@ -653,6 +751,61 @@ $ kubectl create secret generic gem-license -from-file=license.jwt
...
+*Default value: * `gossip-ring`
+
+`memberlist.join` is an address used to find memberlist peers for ring gossip
+
+### string ruler.args.ruler-storage.s3.bucket-name
+
+*Default value: * `ruler`
+
+`ruler-storage.s3.bucket-name` is name of the bucket in which the ruler data will be stored.
```
Comment (MichelHollands) [inline enterprise-metrics/docs/README.md:791]:
```suggestion
`ruler-storage.s3.bucket-name` is the name of the bucket in which the ruler data will be stored.
```

File: enterprise-metrics/docs/README.md (hunk 1 above)
Comment (jdbaldry) [reply enterprise-metrics/docs/README.md:332]: Good catch, these docs are generated from the `main.libsonnet` file using `make docs/README.md` so I will fix it there and regenerate

**Pull request merged**